### PR TITLE
No longer exposing retrieveValueFromStorage() as a public function

### PR DESCRIPTION
**Pull request by LefterisJP**
- plus small fix in EndToEndTests

### CODE REVIEW

**Comment by chriseth** [comment]
Just a heads up: testeth on OSX (buildbot) reported:
```
unknown location:0: fatal error in "state_variable_accessors": signal: SIGSEGV, si_code: 0 (memory access violation at address: 0x00000000)
/Users/administrator/slave/build-cpp-ethereum-osx-pr/build/test/SolidityNameAndTypeResolution.cpp:647: last checkpoint
```

File: libsolidity/BaseTypes.h [hunk 1]
```diff
@@ -49,6 +51,9 @@ struct Location
 /// Stream output for Location (used e.g. in boost exceptions).
 inline std::ostream& operator<<(std::ostream& _out, Location const& _location)
 {
+	if (_location.isEmpty())
+		return _out << "NO_LOCATION_SPECIFIED";
```
Comment (chriseth) [inline libsolidity/BaseTypes.h]: But we know the source name.

File: libsolidity/BaseTypes.h [hunk 2]
```diff
@@ -49,6 +51,9 @@ struct Location
...
+	//else
```
Comment (chriseth) [inline libsolidity/BaseTypes.h]: Superfluous

File: libsolidity/ExpressionCompiler.cpp [hunk 3]
```diff
@@ -834,23 +834,23 @@ void ExpressionCompiler::LValue::retrieveValue(Expression const& _expression, bo
...
 			*m_context << eth::dupInstruction(stackPos + 1);
 		break;
 	}
 	case STORAGE:
-		retrieveValueFromStorage(_expression.getType(), _remove);
+		retrieveValueFromStorage(_type, _remove);
 		break;
 	case MEMORY:
-		if (!_expression.getType()->isValueType())
+		if (!_type)
```
Comment (chriseth) [inline libsolidity/ExpressionCompiler.cpp]: Did you change that on purpose?

Comment (LefterisJP) [inline libsolidity/ExpressionCompiler.cpp]: no, nice catch

**Pull request merged**